### PR TITLE
Adding hiera info for DHCP/DNS

### PR DESCRIPTION
**Pull request by lzap**
@ekohl what is the DNS syntax again? I will add it.

### CODE REVIEW

**Comment by ekohl** [comment]
For DNS you can use all parameters that https://github.com/theforeman/puppet-dns/blob/master/manifests/zone.pp has, similar to how you can use all parameters in https://github.com/theforeman/puppet-dhcp/blob/master/manifests/pool.pp for DHCP.

The ```reverse``` parameter is really badly named and should be ```add_a_record```. If you want a clean master zone example.com with just a ```NS``` record you'd use:
```yaml
dns::zones:
  example.com:
    reverse: true
```

Edit: https://github.com/theforeman/puppet-foreman_proxy/blob/a11b521b2be6216b403766315dde169f1573a340/manifests/proxydns.pp#L31-L41 is how we create the default zones and might give you some idea.

**Comment by lzap** [comment]
Could you please just give me two examples of typical DNS zone and reverse zone I can copy paste?

**Comment by ekohl** [comment]
In the current version you can abuse the ```--foreman-proxy-dns-reverse``` option to create empty zones (including forwarding). So
```
foreman-installer --foreman-proxy-dns-reverse example.com --foreman-proxy-dns-reverse 1.168.192.in-addr.arpa
```

In the next version you should be able to use the ```dns::zones``` parameter as well through hiera. Note the following is untested because nightlies are broken.
```yaml
dns::zones:
  example.com:
    # creates @ SOA $::fqdn root.example.com.
    # creates $::fqdn A $::ipaddress
  example.net:
    # creates @ SOA test.example.net. hostmaster.example.com.
    # creates test.example.net A 192.0.2.100
    soa: test.example.net
    soaip: 192.0.2.100
    contact: hostmaster.example.com.
  example.org:
    # creates @ SOA $::fqdn root.example.org.
    # does NOT create an A record
    reverse: true
  2.0.192.in-addr.arpa:
    # creates @ SOA $::fqdn hostmaster.example.com.
    reverse: true
    contact: hostmaster.example.com.
```

**Comment by lzap** [comment]
So by current version you mean 1.15 and next version is 1.16/develop?

**Comment by lzap** [comment]
Thanks rebased.

**Comment by ekohl** [comment]
> So by current version you mean 1.15 and next version is 1.16/develop?

Correct.

**Comment by ekohl** [comment]
Note that I'd like to test this once nightlies are up & running again.

**Comment by GregSutcliffe** [comment]
@ekohl happy to merge once you've tested it :)

**Comment by ekohl** [comment]
I also wonder if we should refer to some Foreman documentation in https://github.com/theforeman/foreman-installer/blob/develop/config/foreman.hiera/custom.yaml for some common examples.

**Comment by lzap** [comment]
Honestly in config comment sections, I'd expect full and working examples so you can simply uncomment and edit them.

**Comment by ekohl** [comment]
While I usually agree, there are so many options users could want that I don't know what examples we should include.

**Comment by lzap** [comment]
Sure, what do you want me to do again? I am afraid I don't understand what do you want me to "refer to" and where exactly.

**Comment by ekohl** [comment]
For some reason puppet-dns in the nightly builds does not contain the ```dns::zones``` parameter. I'll have to look into why it's stuck on the commit just before it was added.

**Comment by ekohl** [comment]
Now that the nightlies are fixed I tested it and discovered a cyclic dependency when using the parameter. https://github.com/theforeman/puppet-dns/pull/97 should fix that including a test for it.

**Comment by ekohl** [comment]
With that fixed I can now say I suggested some invalid YAML; I'd now suggest:
```yaml
dns::zones:
  # creates @ SOA $::fqdn root.example.com.
  # creates $::fqdn A $::ipaddress
  example.com: {}

  # creates @ SOA test.example.net. hostmaster.example.com.
  # creates test.example.net A 192.0.2.100
  example.net:
    soa: test.example.net
    soaip: 192.0.2.100
    contact: hostmaster.example.com.

  # creates @ SOA $::fqdn root.example.org.
  # does NOT create an A record
  example.org:
    reverse: true

  # creates @ SOA $::fqdn hostmaster.example.com.
  2.0.192.in-addr.arpa:
    reverse: true
    contact: hostmaster.example.com.
```

Other than that I think it's good to go.

**Comment by ekohl** [comment]
Ping

**Comment by lzap** [comment]
Sorry done.

File: plugins/katello/nightly/installation/index.md
Comment (ekohl) [inline plugins/katello/nightly/installation/index.md]: The installer only supports one subnet and one DNS domain via command line arguments.

**Review by ekohl — CHANGES_REQUESTED** [review]
Other than the one sentence I think it looks good.

**Comment by lzap** [comment]
Done, finally thanks for huge help here!

**Review by ekohl — APPROVED** [review]
Anyone with merge permissions: please have a look :)

**Comment by ares** [comment]
Thanks @lzap and @ekohl, merging

**Pull request merged**
